app/controllers/planning_applications/neighbours_controller.rb

```diff
@@ -14,7 +14,13 @@ def index
 
     def create
       begin
-        @consultation.update!(consultation_params)
+        existing_neighbour_addresses = @consultation.neighbours.map(&:address)
```

Comment:
Instead of loading all these records in memory (it could be 100s) can we just do a direct query.
Or i'm probably leaning towards the find_or_create_by approach

Reply:
My concern was that if we queried it each time in the loop that might end up being worse. But that depends a lot on usage.

If there are already a lot of neighbours associated with the consultation, loading them into memory is not great as you say, and if only a couple of addresses are being added then you end up with potentially hundreds stored to avoid only a couple of queries.

But on the other hand if there are none currently associated, and a large area is searched for, then it'll do hundreds of queries, but caching the existing addresses would have stored none in memory.

Perhaps you're right that the find-or-create way is better.